### PR TITLE
Issue 628 included activities

### LINKED ISSUE #628: SAP: included-activities
**Extended Description**
- As an ASAP developer, in order to ensure included activities for testing are identified, I want to test for an included-activities part within the terms and conditions, and included-activity part(s) within the included-activities part.

**Acceptance Criteria**
- [x] All Schematron assertion messages are declarative statements which affirm the positive test outcome.
- [x] All Schematron assertion diagnostic messages are declarative statements which explain the negative test outcome.
- [x] The Schematron code has no assertion failures when validated using `src/validations/styleguides/sch.sch` using the `basic` phase.
- [x] XSpec unit tests for positive and negative Schematron assertion outcomes accompany all Schematron assertions (where feasible).
- [x] Check for part name='included-activities'
- [x] check for part name (at least one) of 'included-activity' as child of part name 'included-activities'

**Definition of Done**
 - [x] Acceptance criteria met
 - [x] Unit test coverage of our code > 95%
 - [x] Automated code quality checks passed
 - [x] Security reviewed and reported
 - [x] Reviewed against plain language guidelines
 - [x] Code must be self-documenting
 - [x] No local tech debt
 - [x] Load/performance tests passed – needs to be created/automated
 - [x] Documentation updated
 - [x] Architectural Decision Record completed as necessary for significant design choices
 - [ ] PR reviewed & approved
 - [ ] Source code merged

### PR DESCRIPTION
**Pull request by markXLIX**
Closes #628 
Added asserts, diagnostics, xspec for:
terms-and-conditions/part[@name='included-activities']
part[@name='included-activities']/part[@name='included-activity']

formatting

### CODE REVIEW

File: src/validations/rules/sap.sch [hunk 1]
```diff
@@ -423,7 +431,8 @@
                 fedramp:specific="true"
                 id="assumption-ordered"
                 role="error"
-                test="deep-equal($unsorted_assumptions, $sorted_assumptions)">The SAP terms and conditions assumptions part has assumption parts that are ordered.</sch:assert>
+                test="deep-equal($unsorted_assumptions, $sorted_assumptions)">The SAP terms and conditions assumptions part has assumption parts that
```
Comment (danielnaab) [inline src/validations/rules/sap.sch:434]:
```suggestion
                test="deep-equal($unsorted_assumptions, $sorted_assumptions)">The SAP terms and conditions assumptions part must have assumption parts that
```

File: src/validations/rules/sap.sch [hunk 2]
```diff
@@ -480,6 +489,20 @@
...
+        <sch:rule
+            context="oscal:terms-and-conditions/oscal:part[@name eq 'included-activities']">
+
+            <sch:assert
+                diagnostics="has-roe-included-activity-diagnostic"
+                doc:guide-reference="Guide to OSCAL-based FedRAMP Security Assessment Plans (SAP) §4.18"
+                fedramp:specific="true"
+                id="has-roe-included-activity"
+                role="error"
+                test="oscal:part[@name eq 'included-activity']">The SAP Rules of Engagement (ROE) included activities have one or more included
```
Comment (danielnaab) [inline src/validations/rules/sap.sch:501]:
```suggestion
                test="oscal:part[@name eq 'included-activity']">The SAP Rules of Engagement (ROE) included activities must have one or more included
```

File: src/validations/test/rules/sap.xspec [hunk 3]
```diff
@@ -1043,192 +1043,238 @@
...
+                    <x:context>
+                        <assessment-plan
+                            xmlns="http://csrc.nist.gov/ns/oscal/1.0" />
+                    </x:context>
+                    <x:expect-assert
+                        id="has-location-assessment-subject"
+                        label="that is incorrect" />
+                </x:scenario>
+                <x:scenario
+                    label="and there is not an include-all child element">
```
Comment (danielnaab) [inline src/validations/test/rules/sap.xspec:1211]:
```suggestion
                    label="when there is not an include-all child element">
```

File: src/validations/test/rules/sap.xspec [hunk 4]
```diff
@@ -1043,192 +1043,238 @@
...
+                                    id="assessor" />
+                            </metadata>
+                        </assessment-plan>
+                    </x:context>
+                    <x:expect-not-assert
+                        id="has-metadata-role-assessor"
+                        label="that is correct" />
+                </x:scenario>
+                <x:scenario
+                    label="there is not a role element with an @id of 'assessor'">
```
Comment (danielnaab) [inline src/validations/test/rules/sap.xspec:1239]:
```suggestion
                    label="when there is not a role element with an @id of 'assessor'">
```

**Review by danielnaab — CHANGES_REQUESTED** [review]
Minor wording suggestions

**Review by GaryGapinski — APPROVED** [review]

**Review by danielnaab — APPROVED** [review]

**Pull request merged**